source/Engine.cpp

```diff
@@ -2194,7 +2194,11 @@ void Engine::DoCollisions(Projectile &projectile)
 		// For weapons with a trigger radius, check if any detectable object will set it off.
 		double triggerRadius = weapon.TriggerRadius();
 		if(triggerRadius)
-			for(const Body *body : shipCollisions.Circle(projectile.Position(), triggerRadius))
+		{
+			vector<Body *> inRadius;
+			inRadius.reserve(ships.size());
```

Comment:
```suggestion
			vector<Body *> inRadius(ships.size());
```
are you going to multithread this too? if not we could have a static

Reply:
I think a static would be overkill here.

Reply:
ye ok